buttonAdd.js

```diff
@@ -0,0 +1,63 @@
+document.getElementById("button").addEventListener("click", function () {
+  document.querySelector(".bg-modal").style.display = "flex";
+});
+
+document.querySelector(".close").addEventListener("click", function () {
+  document.querySelector(".bg-modal").style.display = "none";
+});
+
+let data = [];
+const addProduct = (ev) => {
+  ev.preventDefault(); //to stop the form submitting
+  let product = {
+    id: Date.now(),
+    photo: document.getElementById("photo").value,
+    name: document.getElementById("name").value,
+    category: document.getElementById("category").value,
+    price: document.getElementById("price").value,
+  };
+  data.push(product);
+  // document.forms[0].reset(); // to clear the form for the next entries
+  document.querySelector("form").reset();
+
+  //for display purposes only
+  console.warn("added", { data });
```

Comment:
Never commit with console logs :-), those are for you to see while developing, but they should never go to master :-)